src/device/router_device_worker.erl

```diff
@@ -1621,6 +1621,7 @@ handle_frame_timeout(
                     fcnt = FCntDown,
                     fopts = FOpts1,
                     fport = Port,
+                    adr = Frame#frame.adr,
```

Comment:
I don't understand how this line relates to the other new comment (`ADR bit normally must be set if MAC ADR command is downlinked`) in the test suite.

Reply:
The end-device on the Uplink sets the ADR bit.  On the Downlink Ack we should also set the ADR bit to mirror the setting on the Uplink.

The spec allows the LNS leeway on setting the ADR bit, so this is not an entirely black & white issue fix.  If the Network temporarily cannot estimate the best data rate owing to rapid changes of the radio channel, it clears the ADR bit and thus signals the end-device that the end-device should rely on it's own behavior heuristics, i.e. usually set maximum power.  Our LNS code is probably not that smart yet.

Reply:
Good summary. Either that's new language in the spec or I missed it the last time I studied up on ADR.